google/cloud/internal/curl_impl.cc

```diff
@@ -763,6 +776,10 @@ Status CurlImpl::MakeRequest(CurlImpl::HttpMethod method,
     if (!status.ok()) return OnTransferError(std::move(status));
     status = handle_.SetOption(CURLOPT_UPLOAD, 1L);
     if (!status.ok()) return OnTransferError(std::move(status));
+    status = handle_.SetOption(CURLOPT_INFILESIZE_LARGE,
+                               static_cast<long>(writev.size()));  // NOLINT
```

Comment:
I think this is the number of elements in the vector, not the total number of bytes?  In other words, you are calling (for example) `handle_.SetOption(CURLOPT_INFILESIZE_LARGE, 2)` when you want `handle_.SetOption(CURLOPT_INFILESIZE_LARGE, writev[0].size() + writev[1].size())` ?

Reply:
I had wondered the same thing, but it appears that `WriteVector.size()` is, perhaps suprisingly, not the size of the vector but rather the sum of the sizes over the vector, as you desire.  (In contrast, however, `WriteVector.empty()` is just on the vector.)